### ISSUE TITLE
Repeating last show in series

### ISSUE THREAD

**Issue by cjmanca** [mdhiggins/PlexAutoSkip#14]
Now that we have it skipping to the next episode before the countdown starts, it seems that this causes it to just repeat the same episode over and over if you're at the last episode in the series (rather than just returning to the main menu).

I remember seeing this very rarely before (I assume that it was just a timing issue where it happened to check the show time at the exact right time before the player had returned to the main menu), but now it happens every time at the end of the last episode in the series.

Is there any way to check if there's a next episode via the API before attempting to skip to it?

**Comment by mdhiggins** [comment]
45c8f28c4ee3c6024279f517015084c8a3c0e673

See if that works

This gets a little tricky, not all players return timeline data which is required to get the playqueue. The playqueue is what is needed to determine if you're at the end of the line or not

Its working on my quick tests using the web player but there will be some limitations if the player has buggy timeline returns. I think most of the mainstream players should be ok

**Comment by cjmanca** [comment]
Working on my "Plex For Windows" player, although that's not surprising. I'll test it on FireTV later, thanks.

Do you have any idea what the latency is on that `PlayQueue.get` api call? Unless it's super fast, then I'd be worried about it missing the window where it can skip to next episode prior to the countdown starting.

If so - maybe we could call it earlier when the playing episode is first detected, and cache the result

**Comment by mdhiggins** [comment]
It seems pretty much instant, if you run into any issues I can look about offloading that somewhere earlier but it gets a little tricky as its a subfunction of the player not the media session and there can in theory be multiple players reported so its best handled during the player loop

**Comment by mdhiggins** [comment]
Getting about 6-7 milliseconds for PlayQueue.get()

**Comment by cjmanca** [comment]
Ah, no that should be just fine then, I can't see that being an issue at all at that latency.

**Comment by mdhiggins** [comment]
Going to close this out, let me know if you have any issues

**Comment by cjmanca** [comment]
It still happens most of the time. I've just learned to live with it.

**Comment by mdhiggins** [comment]
Got any logs? Curious to know if the playqueue function is returning data or not

**Comment by cjmanca** [comment]
```
2022-07-19 17:20:43 - __main__ - INFO - Found 118 [173625] (Overlord s04e03 - Episode 3) Speedy media that has reached the end of its playback with viewOffset 1418993 and duration 1420110 with skip-next enabled, will skip to next
2022-07-19 17:20:43 - __main__ - INFO - Seek target is the end of 118 [173625] (Overlord s04e03 - Episode 3) Speedy for Plex for Windows player , going to next
```

But it seems to give that same log output regardless of whether it restarts the episode or just goes back to the main menu. Do I need to change logging settings somehow to get more output?

**Comment by mdhiggins** [comment]
You'll need to turn on debug level logging

https://github.com/mdhiggins/PlexAutoSkip/wiki/Debug-Level-Logging

**Comment by cjmanca** [comment]
For which area? Root, console or file?

**Comment by mdhiggins** [comment]
Depends on where you're looking at the logs, console if you're watching the console output, file if you're pulling from the .log files

**Comment by cjmanca** [comment]
Ahh, gotcha. I was thinking it meant - file to monitor debug info related to reading config/custom files, and console/root related to other types of debug messages. I'll change them to debug and see if I can't get it to happen again.

Also, unrelated - I developed a utility to detect episode credits and insert them into the plex taggings database. It uses audio spectrographic fingerprinting similar to the native plex intro detection. Works well with your plexautoskip script

**Comment by mdhiggins** [comment]
On my quick tests using the Plex for Windows client which looks to be the one you're using, the client fails to report a timeline on whatever media session is created first (not sure why) then works fine on subsequent playbacks. Without the timeline the check fails

**Comment by mdhiggins** [comment]
> Ahh, gotcha. I was thinking it meant - file to monitor debug info related to reading config/custom files, and console/root related to other types of debug messages. I'll change them to debug and see if I can't get it to happen again.
> 
> Also, unrelated - I developed a utility to detect episode credits and insert them into the plex taggings database. It uses audio spectrographic fingerprinting similar to the native plex intro detection. Works well with your plexautoskip script

Very cool, if you care to share I would definitely check that out

**Comment by cjmanca** [comment]
> On my quick tests using the Plex for Windows client which looks to be the one you're using, the client fails to report a timeline on whatever media session is created first (not sure why) then works fine on subsequent playbacks. Without the timeline the check fails

Hmm, ok, that'll help reproduce the bug, since I was having trouble getting it to happen again, haha. But if that's the reason, I'm not sure what can be done to fix it from our end...

**Comment by cjmanca** [comment]
> > Ahh, gotcha. I was thinking it meant - file to monitor debug info related to reading config/custom files, and console/root related to other types of debug messages. I'll change them to debug and see if I can't get it to happen again.
> > Also, unrelated - I developed a utility to detect episode credits and insert them into the plex taggings database. It uses audio spectrographic fingerprinting similar to the native plex intro detection. Works well with your plexautoskip script
> 
> Very cool, if you care to share I would definitely check that out

Very early release. I only just finished it, so it's a bit rough. Currently still scanning my library. Was my weekend project =P

https://github.com/cjmanca/plex-credits-detect

**Comment by mdhiggins** [comment]
I think I found a solution

2573b6844df9c9307da35268a2ff426e8507f747

Overhauled how the script handles players, instead of iterating through all players listed in a session it identifies the active players and handles them on a session/player basis as the updates come from the server

These updates also include the playQueueID so I don't need to poll the timeline for that information anymore which allows me to work around certain players not always reporting timeline data

This did require a large amount of code modification so make sure I didn't break anything but seems like it's all working great in my tests

This should also be more efficient and responsive in many ways compared to the old way of doing things

**Comment by cjmanca** [comment]
Nice! I'll give it a try

**Comment by cjmanca** [comment]
It's been spamming a lot of this:
```
INFO - Seeking Plex for Android (TV) player playing 275 [122778] (Peppa Pig s04e33 - The Little Boat) TV.4d1c322dc805972e-com-plexapp-android from 303577 to 304960
ERROR - Exception, removing from cache to prevent false triggers, will be restored with next sync
Traceback (most recent call last):
  File "C:\PlexAutoSkip\resources\skipper.py", line 201, in _seekTo
    self.seekPlayerTo(mediaWrapper.player, mediaWrapper, targetOffset)
  File "C:\PlexAutoSkip\resources\skipper.py", line 217, in seekPlayerTo
    pq = PlayQueue.get(self.server, mediaWrapper.playQueueID)
  File "C:\Users\<user>\AppData\Local\Programs\Python\Python310\lib\site-packages\plexapi\playqueue.py", line 133, in get
    c = cls(server, data, initpath=path)
  File "C:\Users\<user>\AppData\Local\Programs\Python\Python310\lib\site-packages\plexapi\base.py", line 54, in __init__
    self._loadData(data)
  File "C:\Users\<user>\AppData\Local\Programs\Python\Python310\lib\site-packages\plexapi\playqueue.py", line 66, in _loadData
    self.selectedItem = self[self.playQueueSelectedItemOffset]
  File "C:\Users\<user>\AppData\Local\Programs\Python\Python310\lib\site-packages\plexapi\playqueue.py", line 71, in __getitem__
    return self.items[key]
IndexError: list index out of range
INFO - Found new session 277 [122779] (Peppa Pig s04e34 - The Sandpit) TV.4d1c322dc805972e-com-plexapp-android viewOffset 0 ['dezba8'], sessions: 1
```

**Comment by mdhiggins** [comment]
Hm weird, that's the API throwing an exception just from trying to load the PlayQueue by the ID, won't have much I can do to fix that

I made a small update to at least catch those exceptions

Any particular steps to produce that? I haven't gotten that error in any of my test

ef9101f95343dfced85a49c4bd1ebb3605fd4bfe

**Comment by cjmanca** [comment]
Unfortunately I'm not sure. That would be my niece watching a kids show downstairs, so I wasn't there to see, however she's too young to be doing anything - it just autoplays from episode to episode. It happened for 4 shows in a row before we turned the show off. I don't think there were any errors on screen at the time.

That was being played on a FireStick4k device, if it matters.

**Comment by cjmanca** [comment]
Oh - I just had it do a strange thing I've never seen before though on my Plex For Windows device. It auto-skipped to the end of the episode, but not to the next episode. It went to the 8 second countdown, which fully counted down to 0, at which point plex went to the next episode on it's own.
```
INFO - Found new session 287 [25252] (Star Trek: Deep Space Nine s02e02 - The Circle (2)) Speedy.qrixzse30dc9sbpecpzj4ts1 viewOffset 0 ['cjmanca'], sessions: 0
INFO - Found skippable marker intro for media 287 [25252] (Star Trek: Deep Space Nine s02e02 - The Circle (2)) Speedy.qrixzse30dc9sbpecpzj4ts1 with range 242057-356520 and viewOffset 242694
INFO - Seeking Plex for Windows player playing 287 [25252] (Star Trek: Deep Space Nine s02e02 - The Circle (2)) Speedy.qrixzse30dc9sbpecpzj4ts1 from 242694 to 356520
INFO - Found skippable marker intro for media 287 [25252] (Star Trek: Deep Space Nine s02e02 - The Circle (2)) Speedy.qrixzse30dc9sbpecpzj4ts1 with range 2687813-2736716 and viewOffset 2688681
INFO - Seeking Plex for Windows player playing 287 [25252] (Star Trek: Deep Space Nine s02e02 - The Circle (2)) Speedy.qrixzse30dc9sbpecpzj4ts1 from 2688683 to 2736716
INFO - Found new session 288 [25253] (Star Trek: Deep Space Nine s02e03 - The Siege (3)) Speedy.qrixzse30dc9sbpecpzj4ts1 viewOffset 0 ['cjmanca'], sessions: 1
INFO - Session 287 [25252] (Star Trek: Deep Space Nine s02e02 - The Circle (2)) Speedy.qrixzse30dc9sbpecpzj4ts1 shares player (qrixzse30dc9sbpecpzj4ts1) with new session 288 [25253] (Star Trek: Deep Space Nine s02e03 - The Siege (3)) Speedy.qrixzse30dc9sbpecpzj4ts1, deleting old session 287
```
(first skip is intro, second skip is credits)

**Comment by mdhiggins** [comment]
Hm, how close the end of the episode is the credit skip?

Looks like its skipping to 2736716 but not sure what the duration is

Update:

I have a theory for the above issue, I wonder if your credit skip is very close to the end and then with the offsets the target view Offset is exceeding the duration of the media, I made a few tweaks that might catch this scenario (should be reproducible with that specific episode of deep space nine)

6291808f0608bf5458b94f6b530ddd3c3f6680e2

Another update:

I notice your credit skip is still tagged as "intro", you can probably label that appropriately (IE outro or credits) and then just add the tag in config.ini

**Comment by cjmanca** [comment]
I tried changing the database entry to credits, and no longer got a "Skip" button in the plex interface, unfortunately. It may work with the autoskip of course, but I'd like the skip button to be available too.

**Comment by mdhiggins** [comment]
Hm yeah that's on Plex can't fix that. I saw some references to "outro" in the code so that might be worth a test

**Comment by cjmanca** [comment]
Good to know, I'll give that a try

**Comment by mdhiggins** [comment]
Any progress on the main part of the issue?

**Comment by cjmanca** [comment]
I haven't had a chance to try your latest update yet, sorry. I just downloaded it now though, so it's running. I'll keep an eye on it and do some tests when I have a chance

**Comment by cjmanca** [comment]
Looks like the most recent update fixed the issue with the late episode skips at least. I also haven't seen it repeat an episode yet with the new changes. So far so good!

I tried changing the intro tags to outro, and didn't get a skip button again, unfortunately.

**Comment by mdhiggins** [comment]
In your testing / logs over the past few days are you seeing any of the following message

"Exception trying to get PlayQueue" (debug level logging)

**Comment by cjmanca** [comment]
Oh, I haven't had logging set to debug level, I'll change that and see.

The only strange thing I've noticed is that once it skipped past an episode (left it un-played). Like - it finished episode 10 and went straight to 12. The log did show 11 load up but then immediately proceeded to 12. I'm unsure if that was an issue with the script or plex though, and I've only seen it the once. If it becomes a recurring thing then I'll look into it more.

**Comment by mdhiggins** [comment]
Feel free to post the logs for that and I'll take a look. Maybe the skip fired late and it had already started episode 11 thus skipping it?

**Comment by cjmanca** [comment]
Unfortunately I lost the log of that quite quickly after turning debug mode on. Debug mode puts out so many messages that it rotated through my max log size in about 30 minutes. I think I see which setting to change to increase that max size though which  I think I'll do just to make sure I don't lose a log I want.

**Comment by mdhiggins** [comment]
5f4779a10e37e60ce32f6e7199569ee2abbbfac4

I just removed the exception block that was added earlier I think the playqueueID null check will be enough, if you start getting the error posted here https://github.com/mdhiggins/PlexAutoSkip/issues/14#issuecomment-1189846867 let me know and I'll revert the change

**Comment by cjmanca** [comment]
I just had another strange skip happen. When starting a new episode, it immediately skipped TO the start of the intro, and then a couple of seconds after, it skipped the intro as well.

```
INFO - Seeking Plex for Windows player playing 148 [169926] (Shadows House s01e05 - The Debut) Speedy.qrixzse30dc9sbpecpzj4ts1 from 1332646 to 1419999
INFO - Seek target is the end, going to next
INFO - Found new session 163 [171291] (Shadows House s01e06 - The Garden Labyrinth) Speedy.qrixzse30dc9sbpecpzj4ts1 viewOffset 0 ['cjmanca'], sessions: 2
INFO - Session 148 [169926] (Shadows House s01e05 - The Debut) Speedy.qrixzse30dc9sbpecpzj4ts1 shares player (qrixzse30dc9sbpecpzj4ts1) with new session 163 [171291] (Shadows House s01e06 - The Garden Labyrinth) Speedy.qrixzse30dc9sbpecpzj4ts1, deleting old session 148
DEBUG - Deleting session 148 [169926] (Shadows House s01e05 - The Debut) Speedy.qrixzse30dc9sbpecpzj4ts1, sessions: 1
INFO - Found skippable chapter Intro for media 163 [171291] (Shadows House s01e06 - The Garden Labyrinth) Speedy.qrixzse30dc9sbpecpzj4ts1 with range 0-51051 and viewOffset 27
INFO - Seeking Plex for Windows player playing 163 [171291] (Shadows House s01e06 - The Garden Labyrinth) Speedy.qrixzse30dc9sbpecpzj4ts1 from 27 to 51051
DEBUG - Recent seek successful, server offset update 60000 meets/exceeds target 51051, setting seeking to False
INFO - Found skippable marker intro for media 163 [171291] (Shadows House s01e06 - The Garden Labyrinth) Speedy.qrixzse30dc9sbpecpzj4ts1 with range 52272-140984 and viewOffset 60483
INFO - Seeking Plex for Windows player playing 163 [171291] (Shadows House s01e06 - The Garden Labyrinth) Speedy.qrixzse30dc9sbpecpzj4ts1 from 60483 to 140984
```
I've verified that there's no intro from 0-51051, so I'm not sure why it thought there was:
![image](https://user-images.githubusercontent.com/6020593/180939377-23f4bf5c-30bf-4068-adb9-639fffece73e.png)

**Comment by cjmanca** [comment]
Also receiving exceptions again with the latest update:
```
INFO - Found 175 [122800] (Peppa Pig s03e02 - The Rainbow) TV.4d1c322dc805972e-com-plexapp-android media that has reached the end of its playback with viewOffset 302862 and duration 304080 with skip-next enabled, will skip to next
INFO - Seeking Plex for Android (TV) player playing 175 [122800] (Peppa Pig s03e02 - The Rainbow) TV.4d1c322dc805972e-com-plexapp-android from 302864 to 304080
ERROR - Exception, removing from cache to prevent false triggers, will be restored with next sync
Traceback (most recent call last):
  File "C:\PlexAutoSkip\resources\skipper.py", line 201, in _seekTo
    self.seekPlayerTo(mediaWrapper.player, mediaWrapper, targetOffset)
  File "C:\PlexAutoSkip\resources\skipper.py", line 223, in seekPlayerTo
    pq = PlayQueue.get(self.server, mediaWrapper.playQueueID)
  File "C:\Users\methejuggler\AppData\Local\Programs\Python\Python310\lib\site-packages\plexapi\playqueue.py", line 133, in get
    c = cls(server, data, initpath=path)
  File "C:\Users\methejuggler\AppData\Local\Programs\Python\Python310\lib\site-packages\plexapi\base.py", line 54, in __init__
    self._loadData(data)
  File "C:\Users\methejuggler\AppData\Local\Programs\Python\Python310\lib\site-packages\plexapi\playqueue.py", line 66, in _loadData
    self.selectedItem = self[self.playQueueSelectedItemOffset]
  File "C:\Users\methejuggler\AppData\Local\Programs\Python\Python310\lib\site-packages\plexapi\playqueue.py", line 71, in __getitem__
    return self.items[key]
IndexError: list index out of range
DEBUG - Deleting session 175 [122800] (Peppa Pig s03e02 - The Rainbow) TV.4d1c322dc805972e-com-plexapp-android, sessions: 1
```

**Comment by cjmanca** [comment]
And just had it completely skip a full episode again. It skipped from s01e06 to s01e08. When going back to the season view, it shows s01e07 with the unplayed mark on it. Here's the debug log:
```
2022-07-26 00:56:48 - __main__ - INFO - Found 163 [171291] (Shadows House s01e06 - The Garden Labyrinth) Speedy.qrixzse30dc9sbpecpzj4ts1 media that has reached the end of its playback with viewOffset 1419203 and duration 1420002 with skip-next enabled, will skip to next
2022-07-26 00:56:48 - __main__ - INFO - Seeking Plex for Windows player playing 163 [171291] (Shadows House s01e06 - The Garden Labyrinth) Speedy.qrixzse30dc9sbpecpzj4ts1 from 1419204 to 1420002
2022-07-26 00:56:48 - __main__ - INFO - Seek target is the end, going to next
2022-07-26 00:57:00 - __main__ - INFO - Found new session 176 [170129] (Shadows House s01e08 - In the Palm of His Hand) Speedy.qrixzse30dc9sbpecpzj4ts1 viewOffset 0 ['cjmanca'], sessions: 2
2022-07-26 00:57:00 - __main__ - INFO - Session 163 [171291] (Shadows House s01e06 - The Garden Labyrinth) Speedy.qrixzse30dc9sbpecpzj4ts1 shares player (qrixzse30dc9sbpecpzj4ts1) with new session 176 [170129] (Shadows House s01e08 - In the Palm of His Hand) Speedy.qrixzse30dc9sbpecpzj4ts1, deleting old session 163
2022-07-26 00:57:00 - __main__ - DEBUG - Deleting session 163 [171291] (Shadows House s01e06 - The Garden Labyrinth) Speedy.qrixzse30dc9sbpecpzj4ts1, sessions: 1
2022-07-26 00:57:00 - __main__ - INFO - Found skippable chapter Intro for media 176 [170129] (Shadows House s01e08 - In the Palm of His Hand) Speedy.qrixzse30dc9sbpecpzj4ts1 with range 0-60060 and viewOffset 524
2022-07-26 00:57:00 - __main__ - INFO - Seeking Plex for Windows player playing 176 [170129] (Shadows House s01e08 - In the Palm of His Hand) Speedy.qrixzse30dc9sbpecpzj4ts1 from 524 to 60060
```
It also did the "skip the start of the episode to the intro" thing again on episode 8 immediately after (which can be seen at the bottom of that log snippet)

And then after I went back and watched episode 7, it didn't skip to the next episode (so I got the 8 second count down):
```
INFO - Found skippable marker intro for media 178 [169924] (Shadows House s01e07 - An Incomplete Map) Speedy.qrixzse30dc9sbpecpzj4ts1 with range 1324126-1416386 and viewOffset 1324756
INFO - Seeking Plex for Windows player playing 178 [169924] (Shadows House s01e07 - An Incomplete Map) Speedy.qrixzse30dc9sbpecpzj4ts1 from 1324760 to 1416386
INFO - Found new session 187 [170129] (Shadows House s01e08 - In the Palm of His Hand) Speedy.qrixzse30dc9sbpecpzj4ts1 viewOffset 251000 ['cjmanca'], sessions: 2
INFO - Session 178 [169924] (Shadows House s01e07 - An Incomplete Map) Speedy.qrixzse30dc9sbpecpzj4ts1 shares player (qrixzse30dc9sbpecpzj4ts1) with new session 187 [170129] (Shadows House s01e08 - In the Palm of His Hand) Speedy.qrixzse30dc9sbpecpzj4ts1, deleting old session 178
DEBUG - Deleting session 178 [169924] (Shadows House s01e07 - An Incomplete Map) Speedy.qrixzse30dc9sbpecpzj4ts1, sessions: 1
```
I just checked the duration of episode 7, which is 1420002, so there's only 3.6 seconds between the end of the previous seek and the end of the episode. I'm guessing that it's not "predicting" the new seek spot of the player, but instead waiting on a new update from the API, and I've noticed that takes 5-10 seconds sometimes. So since that's too much for it to immediately skip to the next episode, but possibly less than an API update, it didn't see that the show was so close to the end before it got the API update.

**Comment by mdhiggins** [comment]
Thanks for the detailed logs and information, made this very easy to troubleshoot

### First

I restored the last commit to catch that exception, seems to be an API limitation or Plex bug that will just have to live with but at least it won't throw errors

### Second

The media samples you posted appear to have chapters and one of those chapters is marked with the title `intro`

Note the difference in the logs

```
INFO - Found skippable **chapter** Intro for media 163 [171291] (Shadows House s01e06 - The Garden Labyrinth) Speedy.qrixzse30dc9sbpecpzj4ts1 with range 0-51051 and viewOffset 27

INFO - Found skippable **marker** intro for media 163 [171291] (Shadows House s01e06 - The Garden Labyrinth) Speedy.qrixzse30dc9sbpecpzj4ts1 with range 52272-140984 and viewOffset 60483
```

That's why you're getting the double skip. To address this, you could either clear that chapter data from the files which would be a pain, or I just pushed an update 3522777f76c33bf0666097c4d84f6a510348675b that allows you to include a prefix on your tags that will selectively only apply them to markers versus chapters

If you change to

```ini
[Skip]
tags = m:intro, commercial, advertisement
```

the intro tag will only apply if its a marker and not a chapter. This can be set in config.ini or config.json for fine tuning

### Third

This does look like the issue with the seek target being very close to the end and the API not updating so the script is still waiting for confirmation that the 'seek' has completed. I adjusted things so that if seek targets fall within the durationOffset instead of seeking near the end it will just skip next which should avoid the problem. You'll probably need to increase the durationOffset slightly to catch more of these but that's the only workaround I can think of

7470e65f3c1727df25fd3ae5ef56de1dc2db7072

Let me know if some any of these are not getting flagged as "watched", plex is pretty good about that but the more of the end of the file we skip the more likely that will be

Hopefully that addresses everything

**Comment by cjmanca** [comment]
What about "predicting" the seek? So when you send the seek command, you also pre-emptively adjust the predicted client play time too? I believe you're already using a local tracking of client play times for more accurate seek timing, right? So if you change this local time when sending the seek command (ie. pretend you got an API timing update when sending the seek, which can of course be updated to a more precise time at the next API update)

**Comment by cjmanca** [comment]
Also - may want to add in a check to see if the required modules are installed. I didn't notice that the requirements had changed at first on the update and took a minute to figure out why I was getting a ton of exceptions. I bet most users who update won't realize they have to re-run the pip install command

**Comment by mdhiggins** [comment]
I updated the docker container to also rerun pip install --upgrade if the PAS_UPDATE variable is set to true and I found a small workaround to make PAS backwards compatible with the new change

4f15e90ad431e4b9f3cee37a459b6e4d830ab776

With regards to predicting the seek, this is already done but the issue is you need some kind of confirmation from the server before you fire your next API command otherwise you run the same risk of the API command being delivered too late.  The viewOffset is always tracked and after a seek is fired it assumes the new offset until a server update comes in to say otherwise

This is further complicated by the fact that the API will sometimes send "late" updates after a seek command is sent with inaccurate viewOffsets so you have to filter and disregard those if they don't make sense based on your last sent viewOffset

Regardless, do the 3 above changes fix their respective issues?

**Comment by mdhiggins** [comment]
Looks like there were actually quite a few more issues introduced with 4.12 that weren't immediately apparent

This update should fix them, but it won't be backwards compatible

d507033c81c91c8f2c74b9423ae75591fec57eb8

That update includes a warning if an older API version is detected and blocks connection otherwise

**Comment by cjmanca** [comment]
It doesn't seem to be skipping at all for me anymore. I tried both the "m:intro" and also the normal "intro" options, but plex just plays through the full intro (both first and credits) with no auto skips.

Here's the debug log of the first couple minutes of a show that has an intro sequence right at the start:
```
INFO - Found new session 36 [151245] (Chilling Adventures of Sabrina s02e06 - Chapter Twenty-Six: All of Them Witches) Speedy.qrixzse30dc9sbpecpzj4ts1 viewOffset 0 ['cjmanca'], sessions: 0
DEBUG - Updating session 36 [151245] (Chilling Adventures of Sabrina s02e06 - Chapter Twenty-Six: All of Them Witches) Speedy.qrixzse30dc9sbpecpzj4ts1 viewOffset 8000, old 10170, diff -2170ms (10s since last update)
DEBUG - Updating session 36 [151245] (Chilling Adventures of Sabrina s02e06 - Chapter Twenty-Six: All of Them Witches) Speedy.qrixzse30dc9sbpecpzj4ts1 viewOffset 18000, old 17999, diff 1ms (9s since last update)
DEBUG - Updating session 36 [151245] (Chilling Adventures of Sabrina s02e06 - Chapter Twenty-Six: All of Them Witches) Speedy.qrixzse30dc9sbpecpzj4ts1 viewOffset 28000, old 27998, diff 2ms (9s since last update)
DEBUG - Updating session 36 [151245] (Chilling Adventures of Sabrina s02e06 - Chapter Twenty-Six: All of Them Witches) Speedy.qrixzse30dc9sbpecpzj4ts1 viewOffset 38000, old 38001, diff -1ms (10s since last update)
DEBUG - Updating session 36 [151245] (Chilling Adventures of Sabrina s02e06 - Chapter Twenty-Six: All of Them Witches) Speedy.qrixzse30dc9sbpecpzj4ts1 viewOffset 48000, old 47994, diff 6ms (9s since last update)
DEBUG - Updating session 36 [151245] (Chilling Adventures of Sabrina s02e06 - Chapter Twenty-Six: All of Them Witches) Speedy.qrixzse30dc9sbpecpzj4ts1 viewOffset 58000, old 58006, diff -6ms (10s since last update)
DEBUG - Updating session 36 [151245] (Chilling Adventures of Sabrina s02e06 - Chapter Twenty-Six: All of Them Witches) Speedy.qrixzse30dc9sbpecpzj4ts1 viewOffset 68000, old 68015, diff -15ms (10s since last update)
DEBUG - Updating session 36 [151245] (Chilling Adventures of Sabrina s02e06 - Chapter Twenty-Six: All of Them Witches) Speedy.qrixzse30dc9sbpecpzj4ts1 viewOffset 78000, old 77982, diff 18ms (9s since last update)
DEBUG - Updating session 36 [151245] (Chilling Adventures of Sabrina s02e06 - Chapter Twenty-Six: All of Them Witches) Speedy.qrixzse30dc9sbpecpzj4ts1 viewOffset 88000, old 88007, diff -7ms (10s since last update)
DEBUG - Updating session 36 [151245] (Chilling Adventures of Sabrina s02e06 - Chapter Twenty-Six: All of Them Witches) Speedy.qrixzse30dc9sbpecpzj4ts1 viewOffset 98000, old 97971, diff 29ms (9s since last update)
DEBUG - Updating session 36 [151245] (Chilling Adventures of Sabrina s02e06 - Chapter Twenty-Six: All of Them Witches) Speedy.qrixzse30dc9sbpecpzj4ts1 viewOffset 108000, old 107998, diff 2ms (9s since last update)
DEBUG - Updating session 36 [151245] (Chilling Adventures of Sabrina s02e06 - Chapter Twenty-Six: All of Them Witches) Speedy.qrixzse30dc9sbpecpzj4ts1 viewOffset 113000, old 112812, diff 188ms (4s since last update)
DEBUG - Updating session 36 [151245] (Chilling Adventures of Sabrina s02e06 - Chapter Twenty-Six: All of Them Witches) Speedy.qrixzse30dc9sbpecpzj4ts1 viewOffset 113000, old 113000, diff 0ms (5s since last update)
```

**Comment by mdhiggins** [comment]
Are you on the latest commit?

d507033c81c91c8f2c74b9423ae75591fec57eb8

**Comment by cjmanca** [comment]
Yes, I just updated before trying this

**Comment by cjmanca** [comment]
Also - do you have discord? May make it easier to discuss these things =P

**Comment by mdhiggins** [comment]
psyc#3322

Its working fine for me but definitely need the latest update cause they changed more in the API
